src/advanced/shared/components/HelpModal.tsx

```diff
@@ -0,0 +1,145 @@
+import { useState } from 'react';
+
+interface HelpModalProps {
+  onToggle?: () => void;
+}
+
+const HelpModal = ({ onToggle }: HelpModalProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleToggle = () => {
+    setIsOpen(!isOpen);
```

Comment:
토글이면 약간 이런 것도 좋을 것 같아요
```suggestion
    setIsOpen((isOpen) => !isOpen);
```